analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs

```diff
@@ -89,7 +89,9 @@ public SymbolicValue WithConstraint(SymbolicConstraint constraint)
         Constraints.ContainsKey(typeof(T));
 
     public bool HasConstraint(SymbolicConstraint constraint) =>
-        Constraints.TryGetValue(constraint.GetType(), out var current) && constraint == current;
+        constraint is not null
```

Comment:
This is weird. Why would you ask for a `null` instead of constraint? That sounds like a structural issue with the invocation side that should not happen

Reply:
Because I had the wrong idea. That's why this PR is marked as WIP/Draft.

Reply:
I saw it's a draft. It was an interesting one, so I came in for a discussion about it :)

If I remember correctly, I just left comments, without requesting changes previously.

Reply:
Yes, no worries. It was just unfortunate timing because I had already realized and corrected my mistake locally, but the PR did not reflect that.